src/Generator/SequentialInvoiceNumberGenerator.php

```diff
@@ -49,15 +49,22 @@ public function __construct(
 
     public function generate(): string
     {
-        $invoiceIdentifierPrefix = $this->dateTimeProvider->__invoke()->format('Y/m') . '/';
+        $currentDate = $this->dateTimeProvider->__invoke();
+
+        $invoiceIdentifierPrefix = $currentDate->format('Y/m') . '/';
 
         /** @var InvoiceSequenceInterface $sequence */
         $sequence = $this->getSequence();
 
         $this->sequenceManager->lock($sequence, LockMode::OPTIMISTIC, $sequence->getVersion());
 
+        if ($this->isMonthChanged($currentDate, $sequence->getLastGeneratedAt())) {
+            $sequence->resetIndex();
+        }
+
         $number = $this->generateNumber($sequence->getIndex());
         $sequence->incrementIndex();
```

Comment:
Not connected with this PR, but could use a small improvement - there's no other use case than to call `getIndex()` and `incrementIndex()`  together. What about merging those two into one method `getNextIndex()`?